db/migrate/20210104183527_add_avg_rating_to_recipe.rb

```diff
@@ -0,0 +1,5 @@
+class AddAvgRatingToRecipe < ActiveRecord::Migration[6.0]
```

Comment:
I think that this will have to be "rating" and then we will need to create an average_rating class method? I could be wrong though.

@hashmaster3k @lkriffell @ckccameron

Reply:
We can make it go either way honestly. I think it would be easier to store average_rating or rating in recipe and have a method that will update it when a UserRecipe's rating is updated.

Reply:
Ah ok I am understanding now. So there will be two parts: `recipe.average_rating` and `user_recipe.rating`

And the recipe.average rating will call a method to update this single value based upon all user_recipes with the same recipe_id. Correct?

Reply:
Mostly. I figured rather than calculate the rating every-time a recipe is called, calculate the rating ONLY when a UserRecipe updates it's rating. Thus following this pattern:

1. New Recipe is created -> average_rating is nil/null
2. User purchases recipe - > new UserRecipe is created -> recipe_rating is nil/null
3. User likes recipe and decides to rate -> update UserRecipe.recipe_rating -> this triggers to method to update average_rating for that recipe.

Reply:
Awesome. Sounds great. I'll merge now!